### PR TITLE
[RFR] Add tests for kubevirt VM Clone dialog

### PR DESCRIPTION
**Pull request by rhrazdil**
Adding tests for VM Clone dialog

PR depends on:
https://github.com/openshift/console/pull/2053
https://github.com/openshift/console/pull/2125

### CODE REVIEW

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.migration.scenario.ts [hunk 1]
```diff
@@ -60,19 +62,6 @@ describe('Test VM Migration', () => {
     VM_BOOTUP_TIMEOUT_SECS,
   );
 
-  it(
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/vm.migration.scenario.ts]: why is this test removed?

Comment (rhrazdil) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/vm.migration.scenario.ts]: Decided with @jelkosz to remove this one as it is tested by migrating a VM twice in the following test case

Comment (suomiy) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/vm.migration.scenario.ts]: ok

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.clone.scenario.ts [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+      execSync(`kubectl delete namespace ${nameValidationNamespace}`);
+    });
+
+    it(
+      'Displays warning in clone wizard when cloned vm is running.',
+      async () => {
+        await vm.action(VMACTIONS.START, false);
+        await waitForStatusIcon(statusIcons.starting, PAGE_LOAD_TIMEOUT_SECS);
+        await vm.action(VMACTIONS.CLONE);
+        await browser.wait(
```
[the remark this answers is not in the document]
Comment (rhrazdil) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/vm.clone.scenario.ts]: Done

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.clone.scenario.ts [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+      expect(cloneDialogView.nameHelperMessage.getText()).toMatch(/already used/);
+
+      await cloneDialog.close();
+    });
+  });
+
+  describe('Test cloning settings.', () => {
+    const testVM = getVmManifest('URL', testName, `cloningvm-${getRandStr(5)}`);
+    const vm = new VirtualMachine(testVM.metadata);
+    const clonedVM = new VirtualMachine({
```
[the remark this answers is not in the document]
Comment (rhrazdil) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/vm.clone.scenario.ts]: Added

File: frontend/packages/kubevirt-plugin/integration-tests/tests/utils/mocks.ts [hunk 4]
```diff
@@ -91,25 +91,28 @@ export function getVmManifest(
   name?: string,
   cloudinit?: string,
 ) {
+  const vmname = name || `${provisionSource.toLowerCase()}-${namespace.slice(-5)}`;
```
Comment (vojtechszocs) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/utils/mocks.ts]:
```suggestion
  const vmName = name || `${provisionSource.toLowerCase()}-${namespace.slice(-5)}`;
```

File: frontend/packages/kubevirt-plugin/integration-tests/tests/utils/mocks.ts [hunk 5]
```diff
@@ -91,25 +91,28 @@ export function getVmManifest(
...
       'vm.kubevirt.io/template-namespace': 'openshift',
+      'vm.kubevirt.io/template.revision': '1',
+      'vm.kubevirt.io/template.version': 'v0.6.2',
       'workload.template.kubevirt.io/desktop': 'true',
     },
   };
   const urlSource = {
     http: {
-      url: 'https://download.cirros-cloud.net/0.4.0/cirros-0.4.0-x86_64-disk.img',
+      url: basicVmConfig.sourceURL,
```
Comment (vojtechszocs) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/utils/mocks.ts]:
```suggestion
      url: basicVMConfig.sourceURL,
```

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.actions.scenario.ts [hunk 6]
```diff
@@ -23,56 +26,64 @@ import {
...
   VMACTIONS,
   TABS,
+  VM_IMPORT_TIMEOUT_SECS,
 } from './utils/consts';
 import { VirtualMachine } from './models/virtualMachine';
 
 describe('Test VM actions', () => {
   const leakedResources = new Set<string>();
-  const testVm = getVmManifest('Container', testName);
+  const testVm = getVmManifest('URL', testName);
```
Comment (vojtechszocs) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/vm.actions.scenario.ts]:
```suggestion
  const testVM = getVMManifest('URL', testName);
```

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.clone.scenario.ts [hunk 7]
```diff
@@ -0,0 +1,314 @@
...
+      expect(
+        searchYAML(`vm.kubevirt.io/name: ${vm.name}`, clonedVM.name, clonedVM.namespace, 'vm'),
+      ).toBeTruthy();
+    });
+  });
+
+  describe('Test DataVolumes of cloned VMs', () => {
+    const urlVMManifest = getVmManifest('URL', testName);
+    const urlVM = new VirtualMachine(urlVMManifest.metadata);
+    const cloudInitVmProvisionConfig = {
```
Comment (vojtechszocs) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/vm.clone.scenario.ts]:
```suggestion
    const cloudInitVMProvisionConfig = {
```

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.actions.scenario.ts [hunk 8]
```diff
@@ -23,56 +26,64 @@ import {
...
   afterAll(async () => {
     removeLeakedResources(leakedResources);
   });
 
   describe('Test VM list view kebab actions', () => {
     const vmName = `vm-list-actions-${testName}`;
+    let vm: VirtualMachine;
 
     beforeAll(async () => {
       testVm.metadata.name = vmName;
```
Comment (vojtechszocs) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/vm.actions.scenario.ts]:
```suggestion
      testVM.metadata.name = vmName;
```

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.clone.scenario.ts [hunk 9]
```diff
@@ -0,0 +1,314 @@
...
+      await cloneDialog.selectNamespace(testNameValidationVM.metadata.namespace);
+      await browser.wait(until.presenceOf(cloneDialogView.nameHelperMessage));
+      expect(cloneDialogView.nameHelperMessage.getText()).toMatch(/already used/);
+
+      await cloneDialog.close();
+    });
+  });
+
+  describe('Test cloning settings.', () => {
+    const testVM = getVmManifest('URL', testName, `cloningvm-${getRandStr(5)}`);
```
Comment (vojtechszocs) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/vm.clone.scenario.ts]:
```suggestion
    const testVM = getVMManifest('URL', testName, `cloningvm-${getRandStr(5)}`);
```

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.clone.scenario.ts (hunk 7 above)
Comment (rhrazdil) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/vm.clone.scenario.ts]: Done

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.actions.scenario.ts (hunk 6 above)
Comment (rhrazdil) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/vm.actions.scenario.ts]: Done

File: frontend/packages/kubevirt-plugin/integration-tests/tests/utils/mocks.ts (hunk 5 above)
Comment (rhrazdil) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/utils/mocks.ts]: Done

File: frontend/packages/kubevirt-plugin/integration-tests/tests/utils/mocks.ts (hunk 4 above)
Comment (rhrazdil) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/utils/mocks.ts]: Done

**Pull request merged**